core/src/mender-client.c

```diff
@@ -1027,20 +1045,15 @@ mender_client_update_work_function(void) {
                 mender_log_info("Artifact installation done, rebooting");
                 mender_client_publish_deployment_status(deployment_id, MENDER_DEPLOYMENT_STATUS_REBOOTING);
 
-                /* Save deployment data to publish deployment status after rebooting */
-                if (MENDER_OK != (ret = mender_set_deployment_data(mender_client_deployment_data))) {
-                    mender_log_error("Unable to save deployment data");
-                }
+                /* Save the next state before running the reboot callback --
+                 * if there is an interrupt (power, crash,...) right after,
+                 * it will reboot anyway so after the new boot, reboot
+                 * verification should happen anyway, the callback in that
+                 * state should be able to see if things went well or
+                 * wrong. */
+                set_and_store_state(MENDER_UPDATE_STATE_VERIFY_REBOOT);
                 if ((MENDER_OK == ret) && (NULL != mender_update_module->callbacks[update_state])) {
-                    /* Save the next state before running the reboot callback --
-                     * if there is an interrupt (power, crash,...) right after,
-                     * it will reboot anyway so after the new boot, reboot
-                     * verification should happen anyway, the callback in that
-                     * state should be able to see if things went well or
-                     * wrong. */
-                    mender_storage_save_update_state(MENDER_UPDATE_STATE_VERIFY_REBOOT, mender_update_module->artifact_type);
```

Comment:
I think we still want do do `set_and_store_state()` here iff there is a reboot callback. If there isn't, it will be called by `NEXT_STATE` below anyway. It just wouldn't be called twice in such a case.

Reply:
I moved it outside of the if-statement, but I moved it back now, because it's stored in `NEXT_STATE` as you said